installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -57,6 +58,23 @@ private static StateToolPaths GetPaths()
 
         private static ActionResult _installStateTool(Session session, out string stateToolPath)
         {
+            // Registry info for network errors
+            // This custom action runs as administrator so we have to specifically set
+            // the registry key for the user using their SID in order for the value to
+            // be available in later immediate custom actions
+            string registryKey = string.Format("HKEY_USERS\\{0}\\SOFTWARE\\ActiveState\\{1}", session.CustomActionData["USERSID"], session.CustomActionData["PRODUCT_NAME"]);
+            RegistryValueKind dataType = RegistryValueKind.String;
+            try
+            {
+                Registry.SetValue(registryKey, networkErrorKey, "false", dataType);
+                Registry.SetValue(registryKey, networkErrorMessageKey, "", dataType);
+            } catch (Exception e)
+            {
+                string msg = string.Format("Could not delete network error registry keys. Exception: {0}", e.ToString());
+                session.Log(msg);
+                RollbarReport.Error(msg, session);
+            }
```

Comment:
As we are using this block twice, here and in line 121ff., I would move it into a function.